sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -923,10 +931,13 @@ private void postRegister() {
         revive();
     }
 
-    @Override
-    public void update(Observable observable) {
-        if (observable == planCoordinator) {
-            suppressOrRevive();
-        }
+    private Plan getDeployPlan() {
+        return getDeployPlan(plans).get();
+    }
+
+    private static Optional<Plan> getDeployPlan(Collection<Plan> plans) {
+        return plans.stream()
+                .filter(plan -> plan.isDeployPlan())
```

Comment:
I'm sure there's not a way for it to happen currently, but for safety's sake we ought to assert somewhere that there's no more than one deploy plan defined. Doesn't have to be in this PR.